File: 7 Reverse Integer/solution.cpp
class Solution {
public:
    int reverse(int x) {
        long result = 0;
        int rem = 0;
        long tmp = abs(x);
        while (tmp!=0){
            rem = tmp%10;
            tmp = tmp/10;
            result = result*10+rem;
        }
        if (result!=(int)result)
            return 0;
        return (x>0)?result:(-1*result);
    }
};

File: 7 Reverse Integer/solution.py
class Solution:
    # @return an integer
    def reverse(self, x):
        result = 0
        res = 0
        tmp = abs(x)
        while (tmp!=0):
            (tmp,res) = divmod(tmp,10)
            result = result*10+res
        if result>math.pow(2,31)-1 or result<-1*math.pow(2,31):
            return 0
        return result if x>0 else -1*result
        